pkg/aws/eni/instances.go

```diff
@@ -45,13 +46,18 @@ type EC2API interface {
 	AssociateEIP(ctx context.Context, eniID string, eipTags ipamTypes.Tags) (string, error)
 }
 
+type MetadataAPI interface {
+	GetInstanceMetadata() (string, string, string, string, string, error)
```

Comment:
I'd recommend enriching the function signature here with variable names as it is otherwise not clear at a glance what this function should return as all return values share the same type:


```suggestion
	GetInstanceMetadata() (instanceID, instanceType, availabilityZone, vpcID, subnetID string, err error) 
```

Another option could be to bundle all return values in a single struct so the function signature would become something like `GetInstanceMetadata() (metadata.InstanceMetadata, err)`